docs/examples/howto/add_new_algorithm.rst

```diff
@@ -0,0 +1,174 @@
+
+Configurability
+----------------
+
+It is customary that an algorithm requires configuration parameters.
+For the sake of an example in hypothetical seeding algorithm it could be parameter steering
+which detector layers should be used. 
+The configuration can be provided to an algorithm through an additional class/structure.
+It can be an inner class of the algorithm or it can be external to it.
+One would use definitely external structure if it is shared among several algorithm classes.
+
+For example that is how the configuration object could look like for `MySeedingAlgorithm`:
+
+.. code-block:: cpp
+
+    struct MySeedingConfig {
+        std::vector<int> layers; // layers to use by the seeder
+        float deltaZ;  // the maximum allowed deviation in r-z plane
+    };
+
+.. tip:: It is customary to put the config structures in ``Acts`` namespace.
+
+The algorithm constructor would take `MySeedingConfig` object during the construction in addition 
+to the argument controlling verbosity of diagnostic messages.
+
+.. code-block:: cpp
+
+    MySeedingAlgorithm::MySeedingAlgorithm( Acts::MySeedingConfig cfg, Acts::Logging::Level lvl):
+      ActsExamples::BareAlgortihm("MySeedingAlgorithm", lvl), 
+      m_cfg(std::move(cfg)){...}
+
+
+Python bindings
+---------------
+In order to use an algorithm in standalone ACTS 
+the algorithm and the associated config structure need to be accessible from python.
+For that so called python bindings need to be created using ``pybind11`` library.
+The binding is defined in C++ code in `Examples/Python/src/` directory. 
+There is one source file per category, in this particular case the file to edit would be `TrackFinding.cpp`.
+
+The configuration structure binding would be defined like this:
+
+.. code-block:: cpp
+
+    using Config = Acts::MySeedingConfig;
+    auto c = py::class_<Config>(m, "MySeedingConfig").def(py::init<>()); // this defined the name know in python
+    ACTS_PYTHON_STRUCT_BEGIN(c, Config);
+    ACTS_PYTHON_MEMBER(layers); // makes the layers field accessible in python
+    ACTS_PYTHON_MEMBER(deltaZ); // makes the deltaZ accessible
+    ACTS_PYTHON_STRUCT_END();
+    patchKwargsConstructor(c);
+
+The algorithm class can be made known to python via such binding definition:
+
+.. code-block:: cpp
+
+    auto alg =
+        py::class_<ActsExamples::MySeedingAlgorithm, 
+                   ActsExamples::BareAlgorithm,
+                   std::shared_ptr<ActsExamples::MySeedingAlgorithm>>(
+            mex, "MeSeedingAlgorithm")
+            .def(py::init<const Acts::MySeedingConfg&, Acts::Logging::Level>(), // makes the constructor callable from python
+                 py::arg("config"), py::arg("level")); // defines constructor arguments
+        // other methods can be exposed to python (typically config accessor) 
+
+If bindings are defined correctly (and everything compiles) they can be tested in standalone python session (see section on setting up python) by typing:
```

Comment:
"The bindings can be tested in a standalone python session:"
(I hope we don't have to say "if you did this right and didn't mess anything up" in instructions :D )